### DISCUSSION TITLE
How to access a relationship inside a Model method?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#1917] [Help] [answered]
Let's say I have the following model that represents a team and has restrictions.

```javascript
export default class Team extends BaseModel {
  // --> ATTRIBUTES <--

  @column({ isPrimary: true })
  public id: number

  // ...

  // --> RELATIONSHIPS <--

  @hasOne(() => TeamRestrictions, { foreignKey: 'team_id' })
  public restrictions: HasOne<typeof TeamRestrictions>

  // --> HELPERS <--

  public async updateRestriction(name: string, status: string = 'increment'): Promise<void> {
    let restrictions = await this.related('restrictions').query().first() // <- Does not work
  }
}
```

In Adonis V4, we used to do `let restrictions = await this.restrictions().fetch()`.

**EDIT**: The code above seems to work. I can console log restrictions. But I get the typescript error below. So what is the appropriate way to do it?

> Argument of type 'string' is not assignable to parameter of type 'ExtractModelRelations<this>'.ts(2345)

<img width="516" alt="Screen Shot 2020-11-17 at 11 49 15" src="https://user-images.githubusercontent.com/1658644/99381037-06a6d100-28cb-11eb-95f2-49c6180505ab.png">

**Comment by WeyIin** [comment]
use `this.restrictions`.
```
  public updateRestriction(name: string, status: string = 'increment') {
    let restrictions = this.restrictions
  }
```

Also restrictions have to be preloaded when you want to use `updateRestriction`.
This is how I use it anyways..

**Comment by thetutlage** [comment, marked as answer]
The type error comes directly from a bug in Typescript. https://github.com/microsoft/TypeScript/issues/37778

**Reply by thetutlage** [reply]
You can get around this as follows:

```ts
  public async updateRestriction(name: string, status: string = 'increment'): Promise<void> {
    const self = this as unknown as Team
    let restrictions = await self.related('restrictions').query().first() // <- Does not work
  }
```

**Reply by LeCoupa** [reply]
Thanks! It is working 👍
